bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BType.java

```diff
@@ -133,11 +133,16 @@ public int hashCode() {
     }
 
     public String getName() {
-        return typeName;
+        return typeName == null ? "" : typeName;
     }
 
     public String getQualifiedName() {
-        return pkg == null ? typeName : pkg.toString() + ":" + typeName;
+        String name = getName();
+        if (name.isEmpty()) {
```

Comment:
Instead of calling getName and do the empty name check again can we simply check typeName != null and return here?

Reply:
I think there are places where empty string is explicitly set as the name too. org.wso2.ballerinalang.compiler.semantics.model.types.BType#BType(int, org.wso2.ballerinalang.compiler.semantics.model.symbols.BTypeSymbol)